# all.sample.yml
#
# vars are in groups, both in alpha-numerical order.
#
##

#
# Dashboard
#

cephadm_dashboard: true

cephadm_dashboard_password: admin
cephadm_dashboard_username: admin

cephadm_dashboard_server_addr: >-
  "{{ hostvars[inventory_hostname].ansible_default_ipv4.address }}"
cephadm_dashboard_server_port: 8080
cephadm_dashboard_server_port_ssl: 8443

cephadm_dashboard_ssl_crt: ""
cephadm_dashboard_ssl_key: ""

#
# Firewall
#

cephadm_firewall_firewalld: true

#
# Health
#

cephadm_health_mon_check: ""

#
# Host
#

cephadm_host_packages:
  - ceph-common
  - cephadm
  - docker.io
  - python3-pip
  - python3-virtualenv

cephadm_host_services:
  - docker

#
# Mgr
#

cephadm_mgr_conf: {}
cephadm_mgr_conf_commands: []

#
# Mon
#

cephadm_mon: ""

#
# Network
#

cephadm_network_ip_version: ipv4

cephadm_network_interface_cluster: eno1
cephadm_network_interface_public: "{{ cephadm_network_interface_cluster }}"
cephadm_network_interface_monitor: "{{ cephadm_network_interface_public }}"
cephadm_network_interface_radosgw: "{{ cephadm_network_interface_public }}"

cephadm_network_vip_cluster: ""
cephadm_network_vip_public: ""

#
# OSD
#

cephadm_osd_cpus: 1

# Not sure how to best explain this behavior yet.
#
#   cephadm_osd_devices:
#     <arbitrary group name>
#       block:
#         - /dev/sdb
#       db:
#         - /dev/sdc
#       wal:
#         - /dev/sdd
#
# The error messages are a bit messy, but this should error under any
# type of misconfiguration.
cephadm_osd_devices:
  hdd:
    block:
      - /dev/sdb

# https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#human-to-bytes
cephadm_osd_memory: "4 GB"

cephadm_osd_per_hdd: 1
cephadm_osd_per_nvme: 4
cephadm_osd_per_ssd: 2
